pkg/agent/clib/clib.go

```diff
@@ -0,0 +1,136 @@
+// +build clib
+// Package main deals with ruby / python / php integrations
+package main
+
+import (
+	"C"
+	"fmt"
+	"time"
+
+	"github.com/pyroscope-io/pyroscope/pkg/agent"
+	"github.com/pyroscope-io/pyroscope/pkg/agent/spy"
+	"github.com/pyroscope-io/pyroscope/pkg/agent/upstream/remote"
+	"github.com/pyroscope-io/pyroscope/pkg/config"
+)
+import (
+	"errors"
+
+	"github.com/pyroscope-io/pyroscope/pkg/agent/pyspy"
+	"github.com/pyroscope-io/pyroscope/pkg/util/caps"
+)
+
+func performOSChecks() error {
+	if !caps.HasSysPtraceCap() {
+		return errors.New("if you're running pyroscope in a Docker container,  add --cap-add=sys_ptrace." +
+			"See our Docker Guide for more information: https://pyroscope.io/docs/docker-guide")
+	}
+	return nil
+}
+
+var Sessions = map[int]*agent.ProfileSession{}
```

Comment:
Guys, any idea on how can we close all sessions that would be still open at unexpected exit?

Reply:
Yeah.. I was gonna add a check to see if the process is still alive every 10 seconds or something like that.